src/zoo.js

```diff
@@ -11,57 +11,137 @@ eslint no-unused-vars: [
 
 const data = require('./data');
 
-function animalsByIds(ids) {
-  // seu código aqui
-}
+const animalsByIds = (...ids) =>
+  data.animals.filter(animal => ids.find(id => id === animal.id));
```

Comment:
Fiz de um jeito bem diferente. O seu ficou bem top, achei um pouco confuso só a parte da condição do filter ser o find. Neste caso o filter vai receber algum elemento quando o find achar algo e retornar o animal. Acredito que a condição será true quando tiver algum objeto(no caso o objeto referente ao animal). E quando não achar o objeto retorna false e então o filter não recebe nada. 
Sem dúvida é uma ferramenta muito interessante e pode ser bem util. Aprendendo no CR ;)
O meu eu fiz assim...
Eu fiz um map na array que a função vai receber, e então sei que vou retornar uma array do mesmo tamanho da array que entrou. Para cada id da array eu retorno o objeto completo do animal com aquele determinado id.
Apesar de diferenças também tem semelhanças
```suggestion
const animalsByIds = (...ids) =>
  ids.map(id => data.animals.find(animal => id === animal.id));
```